python/housinginsights/sources/wmata_distcalc.py

```diff
@@ -0,0 +1,258 @@
+        lon = self.lon
+        Nlihc_id = self.nlihcid
+
+        wmata_headers = self.getWmataHeaders()
+
+        params = {'Lat': lat,
+                  'Lon' : lon,
+                  'Radius':str(radiusinmeters)}
+        response = requests.get('https://api.wmata.com/Bus.svc/json/jStops', params=params, headers=wmata_headers)
+        data = response.json()
+
+        for stop in data['Stops']:
+            walkDist = self.getWalkingDistance(lat, lon, str(stop['Lat']), str(stop['Lon']))
+            if walkDist <= radiusinmeters: #within 0.5 miles walking
+                distCsvWriter.writerow((Nlihc_id, 'bus', stop['StopID'], "{0:.2f}".format(self.getMiles())))
+
+class railStation():
+
+    def __init__(self,wmata_api_key,mapbox_api_key):
+        self.wmata_api_key = wmata_api_key
+        self.mapbox_api_key = {'access_token':mapbox_api_key}
+
+    def getWmataHeaders(self):
+        return { 'api_key': self.wmata_api_key}
+
+    def writeRailInfo(self, infoCsvWriter):
+        """Writes all rail station data to a given CSV writer. Returns the railStations json for future processing
+
+           Parameters:
+           infoCsvWriter - csv writer
+           wmata_api_key - api key for wmata REST services
+           """
+        print("Writing RAIL INFO")
+
+        wmata_headers = self.getWmataHeaders()
+
+        railResponse = requests.get("https://api.wmata.com/Rail.svc/json/jStations", headers=wmata_headers)
+        railStations = railResponse.json()['Stations']
+
+        for station in railStations:
+            #delimit list of lines with colon
+            lines = station["LineCode1"] #there is always at least one station
+            for line_code in ["LineCode2", "LineCode3", "LineCode4"]:
+                if station[line_code] != None:
+                    lines += ":" + station[line_code]
+            infoCsvWriter.writerow((station['Code'], 'rail',station['Name'],str(station['Lat']), str(station['Lon']),lines))
+
+        return railStations
+
+    def writeBusInfo(self, infoCsvWriter):
+        """Writes all bus station data to a given CSV writer.
+
+            Parameters:
+            infoCsvWriter - csv writer
+            wmata_api_key - api key for wmata REST services
+            """
+
+        print("Writing BUS INFO")
+
+        wmata_headers = self.getWmataHeaders()
+
+        response = requests.get('https://api.wmata.com/Bus.svc/json/jStops', headers=wmata_headers)
+        data = response.json()
+
+        for stop in data['Stops']:
+
+            lines = ""
+            for route in stop['Routes']:
+                lines = '{}:{}'.format(lines, route)
+            lines = lines[1:] #take off the first :
+
+            infoCsvWriter.writerow((stop['StopID'], 'bus', stop['Name'], stop['Lat'],stop['Lon'], lines)) #, stop_id_or_station_code))
+    
+
+def main(secretsFileName, distOutputFileName,infoOutputFileName):
+    """Writes two csvs: 1 for general bus/rail info, 1 with distances to wmata for projects
```

Comment:
Ok, so this is the big one - basically everything in main() needs to move into the classes above. This way everything we need to get the data is contained here, and we can just say 'for each ApiConn object, get our data' and all the apiConn classes act the same way. A few other comments above align with this too. 

Goal (pseudo code):
```
for api_source in (TaxApiConn,HousingApiConn,CrimeApiConn,WmataInfoApiConn,WmataDistApiConn):
     api_source.get_data
```